2021-04-24 16:12:12,657 ERROR: Exception on /login2.html [GET] [in /home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/app.py:1892]
Traceback (most recent call last):
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/app.py", line 1926, in dispatch_request
    self.raise_routing_exception(req)
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/app.py", line 1908, in raise_routing_exception
    raise request.routing_exception
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/ctx.py", line 350, in match_request
    result = self.url_adapter.match(return_rule=True)
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/werkzeug/routing.py", line 1945, in match
    raise NotFound()
werkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/app.py", line 1816, in handle_user_exception
    return self.handle_http_exception(e)
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/app.py", line 1744, in handle_http_exception
    return handler(e)
  File "/home/ilya-debian/PyBlog/app/errors.py", line 7, in not_found_error
    return render_template('404.html'), 404
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/templating.py", line 140, in render_template
    ctx.app,
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/ilya-debian/PyBlog/app/templates/404.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/home/ilya-debian/PyBlog/app/templates/base.html", line 21, in top-level template code
    <a href="templates/login2.html"">Log test</a>
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/helpers.py", line 370, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/app.py", line 2216, in handle_url_build_error
    reraise(exc_type, exc_value, tb)
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/flask/helpers.py", line 358, in url_for
    endpoint, values, method=method, force_external=external
  File "/home/ilya-debian/PyBlog/env/lib/python3.7/site-packages/werkzeug/routing.py", line 2179, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'login2'. Did you mean 'login' instead?
2021-04-24 16:12:17,041 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 16:12:36,347 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 16:43:57,732 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 16:44:51,967 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 16:50:23,761 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 16:50:45,960 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 16:51:04,787 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 16:52:45,408 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 16:54:24,911 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:03:00,102 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:04:41,227 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:07:46,737 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:08:27,692 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:09:10,213 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:24:43,826 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:25:22,591 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:25:56,920 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:26:45,347 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:32:12,525 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:42:45,179 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:44:06,724 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:50:28,709 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:52:40,404 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:56:38,760 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 17:58:50,023 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:00:24,619 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:04:17,640 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:05:32,689 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:06:37,586 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:07:39,802 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:10:50,159 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:11:35,471 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:12:01,076 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:13:05,784 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:13:39,107 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:14:13,879 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:15:01,555 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:16:09,667 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:17:56,198 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:18:32,869 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:19:24,744 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:21:28,943 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:22:28,195 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:23:20,765 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:26:16,639 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:40:24,285 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:42:19,714 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:44:08,480 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:45:43,047 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:47:03,109 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:50:47,203 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:51:38,446 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 18:53:09,209 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 19:03:53,959 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 19:16:50,564 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 19:22:14,856 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 19:26:44,333 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 19:30:15,650 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
2021-04-24 19:31:56,239 INFO: Microblog startup [in /home/ilya-debian/PyBlog/app/__init__.py:43]
